### PR TITLE
Extract local-cli into a separate npm package

### PR DESCRIPTION
**Pull request by NickGerleman**
This change extracts local-cli into a separate npm package, with infrastructure to split or add more packages at a later time. These packages are scoped to @react-native-windows to make their role as dependencies of RNW clear. 

Packages are versioned using a similar scheme to react-native-windows of 0.0.0-canary.x for master branches, and appropriate preview/release names in stable branches. Packages are still bumped independently. This allows backporting, and creates some level of clarity without forcing users to redownload CLI bits if they don't change. An alternative would be to version in lockstep, which we can pursue if we see confusion.

`promoteRelease` was updated to support version bumping for more packages. Instead of hardcoding a list, we now let packages opt-into release promotion by setting ```promoteRelease: true` in their package.json. This change also exposed a couple of defects around preview version bumping that were fixed. 

It might be worth playing with the boundaries between CLI packages later.

Note that this cannot be checked in until rnbot has permissions to publish to the @react-native-windows npm orgnaization.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5519)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +51,21 @@
     "**/webdriverio": "5.12.1"
   },
   "beachball": {
-    "gitTags": false
+    "gitTags": false,
+    "groups": [
```
Comment (NickGerleman) [inline package.json:55]: Need to play with ChangelogGroupOptions and its effect

File: package.json [hunk 2]
```diff
@@ -50,7 +51,21 @@
...
+      {
+        "name": "React Native Windows Packages",
+        "include": [
+          "packages/@react-native-windows/*",
+          "vnext"
+        ],
+        "disallowedChangeTypes": [
+          "major",
+          "minor",
+          "patch"
```
Comment (NickGerleman) [inline package.json:65]: There's a beachball bug which causes this to infinitely recurse right now https://github.com/microsoft/beachball/issues/364

File: packages/@rnw-scripts/promote-release/src/promoteRelease.ts [hunk 3]
```diff
@@ -124,6 +124,7 @@ async function writeAdoVariables(vars: {[key: string]: any}) {
  * @param version major + minor version
  */
 async function updateBeachballConfigs(release: ReleaseType, version: string) {
+  // TODO before check-in, parse members of the react-native-windows-packages group and apply version changes to those
```
Comment (dannyvv) [inline packages/@rnw-scripts/promote-release/src/promoteRelease.ts:127]: > before check-in [](start = 9, length = 16)

Friendly reminder :)

**Comment by NickGerleman** [comment]
I forgot that we expect AutoLinkedNativeModules to exist at project generation time. The separate autolinking and project generation code makes that harder, since the template is shared. It might be better to do a single CLI package for now.

**Comment by NickGerleman** [comment]
Yes, I think I'm going to change this back to a single package for now. This PR is already quite large, so doing anything more interesting should likely come later.

File: change/@office-iss-react-native-win32-2020-07-23-14-10-52-bumpdeps.json [hunk 4]
```diff
@@ -1,8 +0,0 @@
-{
-  "type": "none",
```
Comment (NickGerleman) [inline change/@office-iss-react-native-win32-2020-07-23-14-10-52-bumpdeps.json:2]: Need to revert removing this when I merge later.

**Pull request merged**